articles/tech-resources/deep-links-self-serve.md

```diff
@@ -0,0 +1,189 @@
+    ```javascript
+    {
+      "applinks": {
+        "apps": [],
+        "details": [
+          {
+            "appID": "<YOUR_APP_ID>",
+            "paths": [
+              "/f/open-in-app/*"
+            ]
+          }
+        ]
+      }
+    }
+    ```
+1. Send your email through SparkPost with click tracking enabled, using your custom tracking domain and link sub-path:
+    ```
+    Your invoice is available <a data-msys-sublink="open-in-app" href="http://www.example.com/invoice/101">here</a>.
+    ```
+1. Update your mobile app to make an HTTP GET request to the SparkPost click tracking service when it handles the incoming click event. There are more details and sample code for this below.
+1. (Optional) Have your app take action based on the original URL which SparkPost returns in its "3xx redirect" HTTP response.
+
+#### Forwarding Clicks From iOS To SparkPost
+
+When an iOS email client recognizes that your recipient has tapped on a universal link, it sends your application delegate an `NSUserActivity`. To allow SparkPost to track this universal link "click" event, you must call the SparkPost click tracking service to tell it about the activity. The click tracker will respond with a "3xx" redirect to the original URL from your email which may be useful to control app behaviour.
+
+Here's a sample `application:continueUserActivity:restorationHandler` method which calls the SparkPost click tracker to register a click event and also retrieves the original URL:
+
+```objc
+- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity
+ restorationHandler:(void (^)(NSArray *))restorationHandler {
+
+    // Ignore non-universal link activities
+    if (![userActivity.activityType isEqualToString:NSUserActivityTypeBrowsingWeb] || userActivity.webpageURL == nil) {
+        return NO;
+    }
+
+    // Create a task to call the SparkPost click tracker, expecting a 3xx redirect to your original URL.
+    // On completion, the handler will receive an NSURLResponse which knows the original target URL.
+    NSURLSessionDataTask* clickTask = [[NSURLSession sharedSession] dataTaskWithURL:[componentsWithURL:userActivity.webpageURL URL] completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        if (response) {
+            NSURL* originalURL = [response URL];
+            if (originalURL) {
+                NSLog(@"Target URL %@", originalURL.absoluteString);
+            }
+        }
+    }];
+
+    // Kick off the task
+    [clickTask resume];
+
+    return YES;
+}
+```
+
+With a little configuration, a single additional handler in your app, and no additional hosting requirements, you can enable and track iOS universal links through your SparkPost account just as if they were any other link.
```

Comment:
This sentence used to be the last in the article. It was a little weak at the time and it's out of place now. Shall we just remove it?